sycl/test-e2e/Matrix/joint_matrix_activation.cpp

```diff
@@ -0,0 +1,19 @@
+//==---------- joint_matrix_activation.cpp  - DPC++ joint_matrix------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: aspect-ext_intel_matrix
+// UNSUPPORTED: gpu-intel-dg2
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Currently, the outlining into an apply function triggers a bug in IGC
```

Comment:
It doesn't look like this test belongs to SYCL repository. It looks more like it should be ocloc test in IGC repository, since it doesn't test any new SYCL API, but rather ability for IGC to handle more IR variants.
So, I suggest move this test to be ocloc test in IGC repo. That will also improve regression catching during IGC development.

Reply:
This test belongs here so real users can find some more useful tests in the suite. 
It also add this new outlining functionality which also impacts the SYCL compiler.

Reply:
If the goal is to provide real usage scenario for the user, I think this repo is a better place: https://github.com/oneapi-src/oneAPI-samples/tree/main/DirectProgramming/C%2B%2BSYCL

I don't see how outlining functionality impacts the SYCL compiler. What was the change in SYCL compiler that helped to resolve the outlining functionality problem? Per what I can see the fix was solely in IGC.
If so, it should be IGC test, not SYCL E2E test.